src/storage/src/source/types.rs

```diff
@@ -351,60 +351,75 @@ pub struct SourcePersistSinkMetrics {
     pub(crate) error_inserts: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
     pub(crate) error_retractions: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
     pub(crate) processed_batches: DeleteOnDropCounter<'static, AtomicU64, Vec<String>>,
+
+    // TODO(guswynn): consider if this metric (and others) should be put into `StorageState`...
+    pub(crate) enable_multi_worker_storage_persist_sink:
+        DeleteOnDropGauge<'static, AtomicI64, Vec<String>>,
 }
 
 impl SourcePersistSinkMetrics {
-    /// Initialises source metrics for a given (source_id, worker_id)
+    /// Initialises source metrics used in the `persist_sink`.
     pub fn new(
         base: &SourceBaseMetrics,
-        shard_id: &mz_persist_client::ShardId,
         source_id: GlobalId,
+        parent_source_id: GlobalId,
```

Comment:
What is a parent source?

Reply:
Its the top-level source when the source is a subsource!